templates/analysis.yml

```diff
@@ -0,0 +1,137 @@
+# External requirements:
+# ----------------------
+# In order to guard secrets, two variable groups must be present in the
+# ADO library and called in the pipeline YAML file before using this
+# template.
+# These variable groups have the following names and contents:
+# - project-registry-sonarcloud: stores sonarcloud variables/secrets
+#   - sonarKeyPrefix: the common prefix string for the sonar keys in the project
+#   - sonarNamePrefix: the common prefix string for the project names
+#   - sonarOrg: the sonar organization
+#   - sonarUrlPrefix: the common prefix string for the url for the results
+# - project-registry-discord: stores discord variables/secrets
+#   - discordAnalysisChannel: the output channel for discord analysis info
+#   - discordAnalysisKey: the webhook key associated with the above channel
+# Example of calling the variable groups:
+#   variables:
+#   - group: project-registry-sonarcloud
+#   - group: project-registry-discord
+#
+# Functionality Caveat: Angular builds not successfully tested
+# ---------------------
+# All code blocks using the angular flag have not been successfully tested.  
+# The code not been removed for three reasons:
+# 1) it functions without a template, and so provides a central record of how
+#    the Angular build tasks function
+# 2) we hope to revisit and debug the errors that prevent the template from
+#    functioning as expected.  As it stands, it appears that the Angular tasks
+#    do not function when a job checks out multiple repositories for some reason
+# 3) We believe that the code in this template is correct, but the error appears
+#    to exist on the side of the calling YAML file and associated setup, therefore
+#    a savvy user may be able to utilize the functionality even if we were not
+#    successful in doing so.
+# Therefore, the code is included with the caveat of not being successfully tested
+
+parameters:
+- name: microservice  # <-- e.g., gateway, account-microservice
+  type: string
+- name: coveragePath  # <-- may override the coverage path if desired
+  type: string
+  default: ''
+- name: workDir       # <-- will need to override with multi-repo subdirectory
+  type: string        #     e.g., ./project-registry-tracking-microservice/
+- name: maven         # <-- maven is the default build method
```

Comment:
what is the alternative build method?

Reply:
This is now clarified by the refactor, so that there is no maven variable, it is the default behavior if angular = false, otherwise it will try to run the build using angular.